core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -135,6 +135,11 @@ private[spark] class BlockManager(
   // Whether to compress shuffle output temporarily spilled to disk
   private val compressShuffleSpill = conf.getBoolean("spark.shuffle.spill.compress", true)
 
+  // Max number of failures before this block manager refreshes the block locations from the driver
+  private val maxFailuresBeforeLocationRefresh =
+    conf.getInt("spark.block.failures.beforeLocationRefresh", Int.MaxValue)
```

Comment:
If we really want to make this constant I think it should be < 10, because with the default retry wait/count settings 10 failures correspond to 15*10 = 2.5 minutes just looping through removed executors.